### PR TITLE
[MONDRIAN-2319] - Calculating Date difference produce wrong result

### PR DESCRIPTION
**Pull request by akhayrutdinov**
- change Interval.diff() for D and Y members to mimic DateDiff function from VBA
- add test cases for DateDiff('y'/'d',...)
- do some cosmetic changes to make affected class follow the checkstyle

@mkambol, @lucboudreau, review it please

### CODE REVIEW

File: doc/vba_functions.html [hunk 1]
```diff
@@ -500,7 +500,7 @@
 	q Quarter <br>
 	m Month <br>
 	y Day of year <br>
```
Comment (lucboudreau) [inline doc/vba_functions.html]: y = Day of year?

Comment (akhayrutdinov) [inline doc/vba_functions.html]: @lucboudreau, yes it is. See it here: https://msdn.microsoft.com/en-us/library/b5xbyt6f%28v=vs.90%29.aspx

**Comment by lucboudreau** [comment]
Just one more question to be sure. I'm asking because a similar situation happened in the past. Was this code taken as a copy from anywhere, or is this original work? I'm merely asking because date-related code tends to be copied around a lot and we must make sure that nothing gets copy-pasted in ours.

**Comment by akhayrutdinov** [comment]
@lucboudreau, I adapted ```computeJdn()``` from here: https://github.com/pentaho/pentaho-vertica-bulkloader/blob/master/src/org/pentaho/di/verticabulkload/nativebinary/ColumnSpec.java#L328
I added there previously while solving another date-related problem. The formula is taken from Wikipedia, it is public domain :)

**Comment by lucboudreau** [comment]
All content on Wikipedia, albeit 'public domain', is released according to the terms of a specific license. 

Let's merge it for now, but I'll check with our IP team. Great job btw.

**Comment by julianhyde** [comment]
I can't believe that I did this wrong (and it's been wrong for all this time!). Thanks for fixing.

If you need some clean code you can copy from Calcite. Apache license, wrote it myself. DateTimeUtils.ymdToJulian.

https://github.com/apache/incubator-calcite/blob/94752a6066cd8a379782510a78a885cffb5aadb9/avatica/src/main/java/org/apache/calcite/avatica/util/DateTimeUtils.java#L708

**Comment by akhayrutdinov** [comment]
@lucboudreau, @mkambol, do I need to add something more? The PR is neither merged, nor declined :)

**Comment by lucboudreau** [comment]
This change broke the build. Please fix ASAP.

**Pull request merged**